src/authenticationservices.cs

```diff
@@ -192,5 +227,481 @@ interface ASWebAuthenticationSession {
+
+		[Export ("httpBody")]
+		NSData HttpBody { get; }
+
+		[Export ("realm")]
+		string Realm { get; }
+
+		[Export ("extensionData")]
+		NSDictionary ExtensionData { get; }
+
+		[Export ("callerBundleIdentifier")]
+		string CallerBundleIdentifier { get; }
+
+		[Export ("authorizationOptions")]
+		NSDictionary AuthorizationOptions { get; }
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationRequest : NSCopying, NSSecureCoding {
+
+		[Export ("provider", ArgumentSemantic.Strong)]
+		IASAuthorizationProvider Provider { get; }
+	}
+
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationSingleSignOnCredential : ASAuthorizationCredential {
+
+		[NullAllowed, Export ("state")]
+		string State { get; }
+
+		[NullAllowed, Export ("accessToken", ArgumentSemantic.Copy)]
+		NSData AccessToken { get; }
+
+		[NullAllowed, Export ("identityToken", ArgumentSemantic.Copy)]
+		NSData IdentityToken { get; }
+
+		[Export ("authorizedScopes", ArgumentSemantic.Copy)]
+		[BindAs (typeof (ASAuthorizationScope []))]
+		NSString[] AuthorizedScopes { get; }
+
+		[NullAllowed, Export ("authenticatedResponse", ArgumentSemantic.Copy)]
+		NSHttpUrlResponse AuthenticatedResponse { get; }
+	}
+
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationSingleSignOnProvider : ASAuthorizationProvider {
+
+		[Static]
+		[Export ("authorizationProviderWithIdentityProviderURL:")]
+		ASAuthorizationSingleSignOnProvider CreateProvider (NSUrl url);
+
+		[Export ("createRequest")]
+		ASAuthorizationSingleSignOnRequest CreateRequest ();
+
+		[Export ("url")]
+		NSUrl Url { get; }
+
+		[Export ("canPerformAuthorization")]
+		bool CanPerformAuthorization { get; }
+	}
+
+	[Introduced (PlatformName.UIKitForMac, 13, 0)]
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (ASAuthorizationOpenIdRequest))]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: -[ASAuthorizationSingleSignOnRequest init]: unrecognized selector sent to instance 0x60000095aa60
+	interface ASAuthorizationSingleSignOnRequest {
+
+		[Export ("authorizationOptions", ArgumentSemantic.Copy)]
+		NSUrlQueryItem[] AuthorizationOptions { get; set; }
+	}
+
+	[TV (13,0), NoWatch, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[Native]
+	enum ASAuthorizationAppleIdButtonType : long {
+		Default = 0,
+		SignUp = 1,
+		Continue = 2,
+	}
+
+	[TV (13,0), NoWatch, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[Native]
+	enum ASAuthorizationAppleIdButtonStyle : long {
+		White = 0,
+		WhiteOutline = 1,
+		Black = 2,
+	}
+
+	[TV (13,0), NoWatch, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (UIControl))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationAppleIDButton {
```

Comment:
You forgot one: `ASAuthorizationAppleIDButton` => `ASAuthorizationAppleIdButton` :smile: